deeplearning4j-modelexport-solr/src/main/java/org/deeplearning4j/nn/modelexport/solr/ltr/model/ComputationGraphLTRScoringModel.java

```diff
@@ -0,0 +1,54 @@
+package org.deeplearning4j.nn.modelexport.solr.ltr.model;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.ltr.feature.Feature;
+import org.apache.solr.ltr.norm.Normalizer;
+import org.deeplearning4j.nn.api.NeuralNetwork;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.util.ModelSerializer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+// TODO: change 7_2 to 7_3 once that is released
+/**
+ * An <a href="https://lucene.apache.org/solr/7_2_0/solr-ltr/org/apache/solr/ltr/model/LTRScoringModel.html">
+ * org.apache.solr.ltr.model.LTRScoringModel</a> that computes scores using a {@link ComputationGraph}.
+ * <p>
+ * Example configuration:
+ * <pre>{
+  "class": "org.deeplearning4j.nn.modelexport.solr.ltr.model.ComputationGraphLTRScoringModel",
+  "name": "myComputationGraph",
+  "features" : [
+    { "name" : "documentRecency", ... },
+    { "name" : "isBook", ... },
+    { "name" : "originalScore", ... }
+  ],
+  "params": {
+    "serializedModelFileName": "mySerializedComputationGraph"
+  }
+}</pre>
+ */
+public class ComputationGraphLTRScoringModel extends NeuralNetworkLTRScoringModel {
+
+  public ComputationGraphLTRScoringModel(String name, List<Feature> features, List<Normalizer> norms, String featureStoreName,
+      List<Feature> allFeatures, Map<String,Object> params) {
+    super(name, features, norms, featureStoreName, allFeatures, params);
+  }
+
+  protected NeuralNetwork restoreNeuralNetwork(InputStream inputStream) throws IOException {
+    return ModelSerializer.restoreComputationGraph(inputStream);
+  }
+
+  @Override
+  public float score(float[] modelFeatureValuesNormalized) {
```

Comment:
You can unify these by the way. They both have a super type of model. You can do some instanceOf checks to see what kind it is. You can also use the ModelGuesser to load any format (it even supports keras and later tensorflow models!)

Reply:
Thanks for the `ModelGuesser` pointer, it makes the changes both simpler and more extensible etc. going forward.